refactor: deprecate `static_url` argument from ModuleSystem [BD-13]

Agrendalath opened the pull request:
## Description

As a part of the ModuleSystem removal, we want to get rid of its `static_url` attribute.

## Supporting information

Sandbox instance: https://pr30046.sandbox.opencraft.hosting/
Use `staff@example.com:edx` to log in.

## Testing instructions

1. This attribute was used only by the CAPA XBlock, so we can test this by checking that [the chemical equation problem](https://app.pr30046.sandbox.opencraft.hosting/learning/course/course-v1:edX+DemoX+Demo_Course/block-v1:edX+DemoX+Demo_Course+type@sequential+block@basic_questions/block-v1:edX+DemoX+Demo_Course+type@vertical+block@vertical_1fef54c2b23b) loads the `chemical_equation_preview.js`.

## Other information

Private-ref: [BB-5518](https://tasks.opencraft.com/browse/BB-5518)

Agrendalath committed:
refactor: deprecate `static_url` argument from ModuleSystem

This argument was officially used only by the ProblemBlock.
If you need to get the base URL for static assets in your XBlock, please use
`settings.STATIC_URL` directly, instead of `runtime.STATIC_URL`.

openedx-webhooks commented:
Thanks for the pull request, @Agrendalath!


When this pull request is ready, tag your edX technical lead.


<!-- comment:welcome-blended -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

Agrendalath commented:
@ormsbee, I extracted this change from #30046.

ormsbee reviewed (changes requested):
Content looks good, but please change the commit message:

1. Since it's removing something, the prefix should be `feat!:`, not `feat:`
2. Please use "remove" instead of "deprecate"–deprecate implies that it's still there but people shouldn't use it because we plan to remove it. This PR actually removes it.
3. Please add something in the commit message explaining what people should use instead.

@jristau1984: I think this would be a good one for merging on Monday. The highest risk would be that static course assets (e.g. images) fail to display properly in XBlocks.

Agrendalath commented:
@ormsbee, thanks for the quick review!

I addressed your points. Just one note:
> Please use "remove" instead of "deprecate"–deprecate implies that it's still there but people shouldn't use it because we plan to remove it. This PR actually removes it.

That's true for the XBlock runtime. Initially, I added a shim to the `ModuleSystemShim`, so it was a bit ambiguous. I added a fixup commit (8dc1099047f422f8d7a2f7ace512ef76ac3a68e4) to revert adding it, so we can have a clean removal scenario here (and a unified experience for both runtimes). Please let me know if you think we should keep it, though.

ormsbee commented:
@Agrendalath: Ack! I totally missed that somehow (for some reason I thought it was being shifted into capa). No, you folks were right, please keep it in, so 2U can monitor to see if anything is still using it before it's finally and completely killed. @jristau1984: The relevant log message to search for in Splunk would be `runtime.STATIC_URL is deprecated. Please use settings.STATIC_URL instead`.

Agrendalath commented:
@ormsbee, I dropped the commit that was removing the XModule shim. For full backward compatibility, I also added back the shim for the XBlock runtime (8f8763a334b6c6fb8561729fd1dec4bdb35d0682), modifying it to show the `DeprecationWarning`.

ormsbee reviewed (approved):
Thank you for your patience. @jristau1984: Please look for the log messages after this has merged:

> `runtime.STATIC_URL is deprecated. Please use settings.STATIC_URL instead`

openedx-webhooks commented:
@Agrendalath 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/30992&entry.1484655318=https://github.com/Agrendalath&entry.752974735=2022-09-14+17:45&entry.1917517419=2022-09-21+16:28&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
